### PR TITLE
Add custom fields documentation

### PR DESCRIPTION
**Pull request by nicolasgnr**
Add a custom field's section in the documentation explaining how to create and assign them to asset versions :)

### CODE REVIEW

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+A custom field is a key-value pair that can be added to an asset version.
+
+Custom fields are managed, which means each asset type has a specific schema defined by an administrator. This enables Exchange to validate the values. For example, an asset can have a custom field configuration named "SupportEndDate," and Exchange can ensure that the value set for this field is a valid date.
+
+Exchange shows custom fields in an asset's overview section:
+
+image::custom-fields.png[]
+
+== Supported types
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Should be Types

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+Exchange supports custom fields of the types enum, number, date, text, list of numbers, and list of strings.
+
+=== Enum
+
+Creating a custom field configuration:
+
+```
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: This should be [source,xml,linenums] for all these code examples

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+curl -X PUT \
+  https://anypoint.mulesoft.com/exchange/api/v1/organizations/:organizationId/assets/:groupId/:assetId/:version/tags/fields/supportCases \
+  -H 'Authorization: bearer <your token here>' \
+  -H 'Content-Type: application/json' \
+  -d '{
+	"tagValue": [1010,2020,3030]
+  }'
+```
+
+=== List of strings
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Should be List of Strings

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+curl -X PUT \
+  https://anypoint.mulesoft.com/exchange/api/v1/organizations/:organizationId/assets/:groupId/:assetId/:version/tags/fields/email \
+  -H 'Authorization: bearer <your token here>' \
+  -H 'Content-Type: application/json' \
+  -d '{
+	"tagValue": “john@doe.com”
+  }'
+```
+
+=== List of numbers
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Should be List of Numbers

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+curl -X PUT \
+  https://anypoint.mulesoft.com/exchange/api/v1/organizations/:organizationId/assets/:groupId/:assetId/:version/tags/fields/maintainers \
+  -H 'Authorization: bearer <your token here>' \
+  -H 'Content-Type: application/json' \
+  -d '{
+	"tagValue": ["John", "Alice", "Fred"]
+  }'
+```
+
+== Restricting Asset Types
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Should be Restrict Asset Types

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+  -d '{
+	"dataType": “enum”,
+	“possibleValues”: [“OAS”,”RAML”],
+	"displayName": “SubType”,
+	"tagKey": “subType”,
+	"assetTypeRestrictions": ["rest-api"]
+  }'
+```
+
+== Deleting A Custom Field From An Asset Version
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Should be Delete a Custom Field from an Asset Version

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+```
+curl -X DELETE \
+  https://anypoint.mulesoft.com/exchange/api/v1/organizations/:organizationId/assets/:groupId/:assetId/:version/tags/fields/:tagKey \
+  -H 'Authorization: bearer <your token here>' \
+  -H 'Content-Type: application/json'
+```
+
+== Deleting A Custom Field
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Should be Delete a Custom Field

**Review by wwarner-salesforce — APPROVED** [review]

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+
+image::custom-fields.png[]
+
+== Supported Types
+
+Exchange supports custom fields of the types enum, number, date, text, list of numbers, and list of strings.
+
+=== Enum
+
+Creating a custom field configuration:
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Are these sections sequential steps that should be in a numbered list?
Also, how does the Custom Fields user get an authorization token?

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 9]
```diff
@@ -0,0 +1,219 @@
...
+  https://anypoint.mulesoft.com/exchange/api/v1/organizations/:organizationId/assets/:groupId/:assetId/:version/tags/fields/subType \
+  -H 'Authorization: bearer <your token here>' \
+  -H 'Content-Type: application/json' \
+  -d '{
+	"tagValue": ["OAS"]
+  }'
+----
+
+=== Number
+
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc:66]: Please supply a description for what a number is in this context. If this field is evaluated numerically, what rules are in place for how to set the minimum, maximum, and default values? What types are allowed (integer, long, double, etc) ?

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+Exchange shows custom fields in an asset's overview section:
+
+image::custom-fields.png[]
+
+== Supported Types
+
+Exchange supports custom fields of the types enum, number, date, text, list of numbers, and list of strings.
+
+=== Enum
+
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc:35]: Please say what an enumerator means for a field. Does this mean you can have multiple pre-defined values like in a drop-down list?

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+  https://anypoint.mulesoft.com/exchange/api/v1/organizations/:organizationId/fields \
+  -H 'Authorization: bearer <your token here>’ \
+  -H 'Content-Type: application/json' \
+  -d '{
+	"dataType": “date”,
+	"displayName": “DueDate”,
+	"tagKey": “dueDate”
+  }'
+----
+
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc:111]: Please explain what the next example is for - Global: Need some explanation before each example that describes what the end result will look like in Exchange and what values to replace.

File: modules/ROOT/pages/to-manage-custom-fields.adoc [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+  -H 'Authorization: bearer <your token here>' \
+  -H 'Content-Type: application/json' \
+  -d '{
+	"tagValue": ["John", "Alice", "Fred"]
+  }'
+----
+
+== Restrict Asset Types
+
+You can restrict a custom field so that it can only be added to certain types of assets. For example, this custom field will only apply to REST APIs:
```
Comment (george-hoffer) [inline modules/ROOT/pages/to-manage-custom-fields.adoc]: Make wording more active: "For example, this custom field only applies to REST APIs:"

**Comment by nicolasgnr** [comment]
@wwarner-salesforce @george-hoffer when do you think this will be merged?

**Comment by wwarner-salesforce** [comment]
George commented, "I'd like to see more info for this doc, such as what these field choices provide in Exchange — for example, does `enum` mean a drop-down list, is there a way to specify a range and a default for `number`? And where do users get the Auth token? Plus we need explanations before each example and a note that the user needs to replace the -d parameters with what they want."

Nico explained that the existing examples show all functionality, and custom fields can be configured and set via API and only viewed via UI, so enums just display the value and no drop-down list, and there is no functionality to set ranges or defaults for number fields.

I added info on auth tokens, explanations for each example, and notes about variables to replace in the examples, including the -d parameters.

PR is ready to review and merge.

**Review by george-hoffer — APPROVED** [review]

**Pull request merged**